systemd/celery.service.d/main.conf

```diff
@@ -0,0 +1,7 @@
+[Service]
+Environment="User=www-data"
```

Comment:
In fact, all of these should likely be set as actual systemd options, not as environment variables:

[Group=](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#User=)
[PidFile=](https://www.freedesktop.org/software/systemd/man/systemd.service.html#PIDFile=)
[WorkingDirectory](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#WorkingDirectory=)

Setting these in environment variables, and then spawning a shell so that you can consume them as environment variables actually leads to less configurability in the long run.

A lot of this could be solved by setting `Type=forking` to let systemd know to just use it's default behavior for a forking process.